docs/concepts/what-is-a-workflow-execution.md

```diff
@@ -2,57 +2,123 @@
+
+### Commands & awaitables
+
+A Workflow Execution does two things:
+
+1. Issue [Commands](/docs/concepts/what-is-a-command)
+2. Wait on an Awaitables (often called Futures)
+
+![Command generation and waiting](/diagrams/workflow-execution-progession-simple.svg)
+
+Commands are issued and Awaitables provided by the use of Workflow APIs in the [Workflow Definition](/docs/concepts/what-is-a-workflow-definition).
+
+Commands are generated whenever the Workflow Function is executed.
+The Worker Process supervises the Command generation and makes sure that it maps to the current Event History.
+(For more information, see [Deterministic constraints](/docs/concepts/what-is-a-workflow-definition/#deterministic-constraints).)
+The Worker Process batches the Commands and then suspends progress to send the Commands to the Cluster whenever the Workflow Function reaches a place where it can no longer progress without a result from a Awaitable.
 
-A Workflow Execution has exclusive access to its local state, executes concurrently to all other Workflow Executions, and can communicate with other Workflow Executions using Signals.
+Awaitables are provided when using APIs for the following:
 
-A Workflow Execution is either Running or Closed.
-When a Workflow Execution is Running, it is either actively progressing or suspended, awaiting on something.
+- Spawning a [Child Workflow Execution](/docs/concepts/what-is-a-child-workflow-execution)
+- Sending a [Signal](/docs/concepts/what-is-a-signal)
+- Spawning an [Activity Execution](/docs/concepts/what-is-an-activity-execution)
+- Requesting cancellation of another Workflow Execution
+- Starting a Timer
+- Awaiting
 
-![Workflow Execution Running status](/diagrams/workflow-execution-running-status.svg)
+A Workflow Execution may only ever block progress on an Awaitable that is provided through a Temporal SDK API.
 
-A Closed status means that the Workflow Execution has finished progressing, and has either Completed successfully, Continued As New, Failed, Timed Out, been Canceled, or Terminated.
+### Status
+
+A Workflow Execution can be either Open or Closed.
 
 ![Workflow Execution statuses](/diagrams/workflow-execution-statuses.svg)
 
-A Workflow Execution is uniquely identified by its [Namespace](/docs/concepts/what-is-a-namespace), [Workflow Id](/docs/concepts/what-is-a-workflow-id), and [Run Id](/docs/concepts/what-is-a-run-id).
+**Open**
 
-The Workflow Id can be used to create a 1:1 mapping between a Workflow Execution and some other resource, such as a customer Id, order Id, or host Id.
+- Running: The only Open status for a Workflow Execution, when Running the Workflow Execution is when a Workflow Execution is either actively progressing or suspended because it is waiting on something.
```

Comment:
```suggestion
- Running: The only Open status for a Workflow Execution.
When Running the Workflow Execution is either actively progressing or suspended because it is waiting on something.
```